### PR TITLE
Add new per-check documentation pages and check severity parameter

### PR DESCRIPTION
**Pull request by katy-sadowski**
This pull request contains the following changes:

- A new `severity` parameter is added to the check descriptions file and to check documentation, as decided in the Fall 2023 DQD Hackathon. This parameter is not used for anything yet, but in the future we will add the ability to specify which severity level(s) to execute in a DQD run
- A new dropdown menu and individual check documentation pages have been added to the DQD website. Docs for 8 checks have been added so far - those with finalized DQD Hackathon worksheets (see [here](https://ohdsiorg.sharepoint.com/:x:/s/Workgroup-Dataquality/EXF1gbCJkxhNtovc5UtYgwkBw2w1Wtmv5Kh1G78JAPqk7A?e=AwKIlh))
- Related updates to documentation configuration files, as well as some cleanups to our vignette headers

Notes for reviewers:

- You may of course ignore all auto-generated files
- I'd recommend to pull down my branch and run `pkgdown::build_site()` so you can review the actual webpage rather than the markdown files
- So far, I only populated the severity parameter for the 8 checks we finalized worksheets for. As we complete the worksheets for more checks, we can open new PRs to add the docs and populate severity in the check description files. (@MaximMoinat I'm thinking you can add docs for your 3 new checks, once we agree on the format here and merge in this PR 😄)
- For now, I'm keeping the original Check Type Descriptions documentation page - we can keep this around until we have docs for all the checks

### CODE REVIEW

**Review by clairblacketer — APPROVED** [review]
This looks great, thank you Katy for putting these together!

File: vignettes/fkClass.Rmd [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+## fkClass
+
+**Check Name**: fkClass
+<br>**Level**: Field check
+<br>**Context**: Verification
+<br>**Category**: Conformance
+<br>**Subcategory**: Computational
+<br>**Severity**: CDM convention &#x26A0;
+
+**Description**: The number and percent of records that have a value in the **cdmFieldName** field in the **cdmTableName** table that do not conform to the **fkClass** class.
```
Comment (MaximMoinat) [inline vignettes/fkClass.Rmd:20]: I propose to make these bolded words into h2 headings (i.e. `## Description`)

File: vignettes/fkClass.Rmd [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  '@cdmTableName.@cdmFieldName' AS violating_field, 
+  co.concept_class_id AS violating_class,
+	cdmTable.* 
+FROM @cdmDatabaseSchema.@cdmTableName cdmTable
+	LEFT JOIN @vocabDatabaseSchema.concept co ON cdmTable.@cdmFieldName = co.concept_id
+WHERE co.concept_id != 0 
+  AND (co.concept_class_id != 'ingredient') 
+```
+
+*ETL Developers*\
```
Comment (MaximMoinat) [inline vignettes/fkClass.Rmd:47]: I propose to make these itelic words into h3 headings (i.e. `### ETL Developers`)

**Review by MaximMoinat — COMMENTED** [review]
Looks good in general. I have some small suggestions for the template, see inline comments on the fkClass page.

**Comment by MaximMoinat** [comment]
@katy-sadowski I have applied a new markdown template to the existing checks, and instantiated the template for all other checks as well.
Other changes:
 - check description files moved to vignettes/checks
 - the template is under extras/checkDescriptionTemplate.Rmd, which is instantiated with createCheckDoc.R
 - completely rebuild the docs folder, which removes some redundant files (e.g. the `_docs` folder)

File: vignettes/checks/plausibleAfterBirth.Rmd [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+The number and percent of records with a date value in the **cdmFieldName** field of the **cdmTableName** table that occurs prior to birth.
+
+## Definition
+This check verifies that events happen after birth. This check is only run on fields where the **PLAUSIBLE_AFTER_BIRTH** parameter is set to **Yes**. The birthdate is taken from the `person table`, either the `birth_datetime` or composed from `year_of_birth`, `month_of_birth`, `day_of_birth` (taking 1st month/1st day if missing).
+
+- *Numerator*: The number of records with a non-null date value that happen prior to birth.
+- *Denominator*: The total number of records in the table with a non-null date value.
+- *Related CDM Convention(s)*: -Not linked to a convention-
+- *CDM Fields/Tables*: By default, this check runs on all date and datetime fields
+- *Default Threshold Value*: 1
```
Comment (katy-sadowski) [inline vignettes/checks/plausibleAfterBirth.Rmd:29]: i think it'd be clearer if this was 1%.  (realize the others are just "0" - maybe those could also be updated to be 0%?)

File: vignettes/checks/plausibleAfterBirth.Rmd [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+## User Guidance
+There might be valid reasons why a record has a date value that occurs prior to birth. For example, prenatal observations might be captured or procedures on the mother might be added to the file of the child. Therefore, some failing records are expected and the default threshold of 1% accounts for that.
+
+However, if more records violate this check, there might be an issue with incorrect birthdates or events with a default date. It is recommended to investigate the records that fail this check to determine the cause of the error and set proper dates. If it is impossible to fix, then implement one of these:
+
+ - Aggressive: Remove all patients who have at least one record before birth (if the birthdate of this patient is unreliable)
+ - Less aggressive: Remove all rows that happen before birth. Probably this should be chosen as a conventional approach for data clean up (if the event dates are unreliable)
+ - Conservative: set event date to birth date.
```
Comment (katy-sadowski) [inline vignettes/checks/plausibleAfterBirth.Rmd:38]: maybe should also add the option of leaving as-is, i.e. for prenatal or other observations where the date is actually valid.

File: vignettes/checks/plausibleAfterBirth.Rmd [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+```sql
+SELECT p.birth_datetime, cdmTable.*
+FROM @cdmDatabaseSchema.@cdmTableName cdmTable
+JOIN @cdmDatabaseSchema.person p ON cdmTable.person_id = p.person_id
+WHERE cdmTable.@cdmFieldName < p.birth_datetime, 
+```
+or, when birth_datetime is missing:
+```sql
+SELECT p.birth_datetime, cdmTable.*
```
Comment (katy-sadowski) [inline vignettes/checks/plausibleAfterBirth.Rmd:54]: change to year, month, day columns

File: vignettes/checks/plausibleAfterBirth.Rmd [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+            '01'
+        )
+    ) AS DATE)
+```
+
+Also, the length of the time interval between these dates might give you a hint of why the problem appears.
+```sql
+select date_difference, count(*)
+from (
+    select daydiff(
```
Comment (katy-sadowski) [inline vignettes/checks/plausibleAfterBirth.Rmd:74]: should be DATEDIFF

File: vignettes/checks/plausibleAfterBirth.Rmd [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        day, 
+        @cdmFieldName, 
+        COALESCE(
+            CAST(p.birth_datetime AS DATE),
+            CAST(CONCAT(p.year_of_birth,'-01-01') AS DATE))
+        ) as date_difference
+    from @cdmTableName ct
+    join person p on ct.person_id = p.person_id 
+) cte
+where date_difference < 0
```
Comment (katy-sadowski) [inline vignettes/checks/plausibleAfterBirth.Rmd:84]: should be > 0 ?  i think datediff does second date - first date

File: inst/csv/OMOP_CDMv5.2_Check_Descriptions.csv [hunk 8]
```diff
@@ -1,27 +1,27 @@
...
+FIELD,plausibleTemporalAfter,The number and percent of records with a value in the @cdmFieldName field of the @cdmTableName that occurs prior to the date in the @plausibleTemporalAfterFieldName field of the @plausibleTemporalAfterTableName table.,Verification,Plausibility,Temporal,field_plausible_temporal_after.sql,plausibleTemporalAfter=='Yes',
+FIELD,plausibleDuringLife,"If yes, the number and percent of records with a date value in the @cdmFieldName field of the @cdmTableName table that occurs after death.",Verification,Plausibility,Temporal,field_plausible_during_life.sql,plausibleDuringLife=='Yes',
+FIELD,withinVisitDates,The number and percent of records not within one week on either side of the corresponding visit occurrence start and end date,Verification,Conformance,,field_within_visit_dates.sql,withinVisitDates=='Yes',
+FIELD,plausibleAfterBirth,"The number and percent of records with a date value in the @cdmFieldName field of the @cdmTableName table that occurs prior to birth.",Verification,Plausibility,Temporal,field_plausible_after_birth.sql,plausibleAfterBirth=='Yes',
```
Comment (katy-sadowski) [inline inst/csv/OMOP_CDMv5.2_Check_Descriptions.csv:23]: @MaximMoinat you can add the severity of your new checks here too!  (and to 5.3 and 5.4 csvs)

**Review by katy-sadowski — COMMENTED** [review]
Thank you so much for these additions @MaximMoinat !  The template looks great and is super useful.  I left some minor suggestions on the plausible after birth doc.

As mentioned in Teams I will wait to merge this until @dimshitc 's new checks are merged in, to save him merge conflict grief 😄

File: vignettes/checks/plausibleAfterBirth.Rmd (hunk 4 above)
Comment (MaximMoinat) [reply vignettes/checks/plausibleAfterBirth.Rmd:38]: I actually like that better for the conservative strategy. Modifying the event date might create other issues. Either it is correct and you leave it as is, or you don't trust the record and remove it, or you don't trust the person and remove it.

File: vignettes/checks/plausibleAfterBirth.Rmd (hunk 7 above)
Comment (MaximMoinat) [reply vignettes/checks/plausibleAfterBirth.Rmd:84]: Indeed! Thanks

File: vignettes/checks/plausibleAfterBirth.Rmd [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    toc: yes
+---
+
+## Summary
+
+**Level**: Field check\
+**Context**: Verification\
+**Category**: Plausibility\
+**Subcategory**: Temporal\
+**Severity**: Characteristic
```
[the remark this answers is not in the document]
Comment (katy-sadowski) [reply vignettes/checks/plausibleAfterBirth.Rmd]: Haha, they are definitely not necessary.  I was thinking maybe in the new report view we build eventually, having some visual marker for the severity could be helpful, but that's something to be discussed later on as we think through the UI.

File: vignettes/checks/plausibleAfterBirth.Rmd [hunk 10]
```diff
@@ -26,16 +26,16 @@ This check verifies that events happen after birth. This check is only run on fi
...
 
 However, if more records violate this check, there might be an issue with incorrect birthdates or events with a default date. It is recommended to investigate the records that fail this check to determine the cause of the error and set proper dates. If it is impossible to fix, then implement one of these:
 
- - Aggressive: Remove all patients who have at least one record before birth (if the birthdate of this patient is unreliable)
- - Less aggressive: Remove all rows that happen before birth. Probably this should be chosen as a conventional approach for data clean up (if the event dates are unreliable)
- - Conservative: set event date to birth date.
+ - Aggressive: Remove all patients who have at least one record before birth (if the birthdate of this patient is unreliable).
+ - Less aggressive: Remove all rows that happen before birth. Probably this should be chosen as a conventional approach for data clean up (if the event dates are unreliable).
+ - Conservative: either keep the records as is (if the date is actually valid, for e.g. prenatal observations).
```
Comment (katy-sadowski) [inline vignettes/checks/plausibleAfterBirth.Rmd:38]: should there be another option after this "either"?

File: vignettes/checks/plausibleAfterBirth.Rmd [hunk 11]
```diff
@@ -14,7 +14,7 @@ output:
 **Context**: Verification\
 **Category**: Plausibility\
 **Subcategory**: Temporal\
-**Severity**: Characteristic
+**Severity**: Characteristization &#10004;
```
Comment (katy-sadowski) [inline vignettes/checks/plausibleAfterBirth.Rmd:17]: typo in characterization 🙃

**Review by katy-sadowski — COMMENTED** [review]
just 2 minor things

**Pull request merged**